super-csv/src/main/java/org/supercsv/io/Tokenizer.java

```diff
@@ -228,32 +231,59 @@ else if( c == quoteChar ) {
 					potentialSpaces = 0;
 					currentColumn.append(c);
 				}
-				
+
 			} else {
-				
+
 				/*
 				 * QUOTE_MODE (within quotes).
 				 */
-				
+
 				if( c == quoteChar ) {
```

Comment:
What happens when the escape char is `\` (i.e. different to the quote char) but double quotes are encountered?

Looks like you're currently handling the normal double quote escaping first, otherwise checking for the custom escape. 

Technically you should leave quote mode when encountering a non-escaped quote, so a double quote isn't valid when using `\` as the escape char.

I'd expect the `c == quoteEscapeChar` block to appear first, and for the `c == quoteChar` block to appear after (i.e. swap the order) and for the `c == quoteChar` block to just reset the state back to `NORMAL` (it doesn't need to check for escaped quotes as that's already been done).  It might mean things get weird if someone uses double quotes though...but that's their own fault for not escaping!

Thoughts?